src/generics.rs

```diff
@@ -32,24 +31,56 @@ pub struct Shared {
     pub peers: HashMap<SocketAddr, Tx>,
 }
 
-#[derive(Getters, Clone)]
```

Comment:
`Getters` simply masked pointers as functions. Doesn't gain you much, and in most cases would _add_ to the weight of your program because it has to create all the code for the functionality.

Simply means that you have to prepend the `&` to something you wish to use as a pointer. Bit cleaner and allows for IDEs that can show type inference to actually display the inferred type.